### ISSUE TITLE
new last.fm beta site rejecting my mpdas login

### ISSUE THREAD

**Issue by pcwalden** [hrkfdn/mpdas#22]
A day or two ago I was forced to the new last.fm beta site and my mpdas scrobbling stopped working.

I have not changed my .mpdasrc. Are they using something other than an MD5 hashed password now? I also noticed I had to use my email address to login to last.fm, not my original last.fm user name. I tried switching to my email address in the mpdasrc, but that did not work either.

>$ /usr/local/bin/mpdas
>(15:30:12) [INFO] Config file (/usr/local/etc/mpdasrc) does not exist or is not readable.
>(15:30:12) [INFO] Connected to MPD.
>(15:30:16) [ERROR] Code: 4
>(15:30:16) [ERROR] Authentification failed. Please check your login data.
>(15:30:16) [INFO] Closing mpdas.

Anyone has issues with the new site?

**Comment by drrok** [comment]
I have the same problem:

~ ❯❯❯ mpdas                                                                                   
(19:02:42) [ERROR] You are not root. Not changing user ..                             
(19:02:42) [INFO] Connected to MPD.                                                              
(19:02:46) [ERROR] Code: 4                                                                      
(19:02:46) [ERROR] Authentication failed. Please check your login data.                         
(19:02:46) [INFO] Closing mpdas.

**Comment by mkody** [comment]
They're doing some migrations, so it's down for... some days?

**Comment by hrkfdn** [comment]
Yeah, it seems to be due to the migration. I can't scrobble from other clients, either.

**Comment by hrkfdn** [comment]
http://status.last.fm/ seems to confirm this.

**Comment by hrkfdn** [comment]
Are you still having problems?

**Comment by drrok** [comment]
still 
![2015-08-27-152457_711x177_scrot](https://cloud.githubusercontent.com/assets/5803407/9521543/ca6cbe3e-4ccf-11e5-98fa-f6c227f48d29.png)

**Comment by hrkfdn** [comment]
Okay thanks, I'll take a look at it later.

**Comment by pcwalden** [comment]
My account is now scrobbling correctly

**Comment by drrok** [comment]
@pcwalden - you rehashed passwd again or just do nothing?

**Comment by pcwalden** [comment]
I did nothing, it just started working again.

**Comment by hrkfdn** [comment]
I can not confirm this problem, either. Can you please double check your configuration?

**Comment by noinia** [comment]
I'm having the same issues. mpdas gives me: 

```
(15:41:57) [ERROR] Code: 4
(15:41:57) [ERROR] Authentication failed. Please check your login data.
```

Scrobbling from [herrie](https://github.com/EdSchouten/herrie) works fine though, with the same credentials.

**Comment by drrok** [comment]
ill check my configuration :) i rehashed psswd again, with this command: 
```
echo -n "psswd" | md5sum
```
still not working

**Comment by mkody** [comment]
Doesn't work for me neither.
Also, when using [Scrobb](https://play.google.com/store/apps/details?id=fr.outadev.lastfm.scrobb) from my phone, it doesn't scrobble directly. Maybe Last.FM still has issues with scrobbling?

**Comment by loljewsandniggersdying** [comment]
I'm having the same issue, had it since weeks, has not resolved itself. 

    —— — (11:07:16) [INFO] Config file (/etc/mpdasrc) does not exist or is not readable.
    (11:07:17) [ERROR] Code: 4
    (11:07:17) [ERROR] Authentication failed. Please check your login data.
    
    [1]+  Exit 1                  mpdas

I use the $HOME/.mpdasrc file which always worked file even though it always gave that line above as error message. It used the home one just file.

**Comment by hrkfdn** [comment]
Since the migration isn't complete yet, I'd blame that. The protocol hasn't changed, mpdas worked before and it still works for some people. I don't think mpdas is the issue here..

**Comment by pcwalden** [comment]
Still working for me.

**Comment by agjacome** [comment]
My last.fm username has uppercase letters. If I use it as is in mpdas config the login fails as described in the issue, but if I put it all lowercased mpdas can log in successfully. Can anyone else of the people having this issue confirm that this uppercase/lowercase nonsense is the cause?

Anyway, it looks that this is not exactly an mpdas issue, since the login is rejected directly by the Last.fm API (small bash test here: https://gist.github.com/agjacome/e7cbf1c5384247cd4b2c **). I don't know where to look to see if this is a documented change or if it should be reported to them.

** Sorry for using mpdas key & secret for this, I am not able to create an API account at the moment, something else is probably broken in their service.

**Comment by noinia** [comment]
Jep, after changing the upercased letters in my username to lowercased ones in mpdasrc I am able to authenticate and scrobble.

**Comment by hrkfdn** [comment]
Glad to hear that! I wish this would have been documented, but I'm assuming it's just a temporary issue due to the migration to the new site. I will merge the patch submitted by agjacome. Thanks everyone!

**Comment by clasick** [comment]
Just a heads up, the same bug is present in mpdas 0.4.3-1. 

The fix is the same as @agjacome mentioned.

**Comment by hrkfdn** [comment]
Mhh, that's a little odd. The fix by @agjacome should be merged since 0.4.0..
